### ISSUE TITLE
Do not register ts-node-config/tsconfig.json in all circumstances?

### ISSUE THREAD

**Issue by dsherret** [cevek/ttypescript#9]
Hey, I was trying to get this to work with istanbul (nyc), mocha, and ts-node. This is my setup (setup is from [here](https://istanbul.js.org/docs/tutorials/typescript/)):

```
// package.json under "scripts"
"test": "cross-env TS_NODE_COMPILER=\"ttypescript\" nyc --reporter=lcov mocha --opts mocha.opts",

// mocha.opts
--require ts-node/register
--recursive
--reporter progress
src/tests/**/*.ts
```

Which gives the following error:

```
V:\test-project\node_modules\ts-node\src\index.ts:209
  const readFile = options.readFile || ts.sys.readFile
                                              ^
TypeError: Cannot read property 'readFile' of undefined
    at Object.register (V:\test-project\node_modules\ts-node\src\index.ts:209:47)
    at Object.<anonymous> (V:\test-project\node_modules\ttypescript\lib\patch.js:5:20)
    at Module._compile (module.js:652:30)
    at Module.replacementCompile (V:\test-project\node_modules\nyc\node_modules\append-transform\index.js:58:13)
    at module.exports (V:\test-project\node_modules\nyc\node_modules\default-require-extensions\js.js:8:9)
   ...etc...
```

I took a look at the line where there's the error and added two console.log statements before:

```ts
// node_modules/ts-node/dist/index.js
var ts = require(compiler);
var transformers = options.transformers || undefined;
console.log(compiler); // outputs: "ttypescript"
console.log(ts); // outputs: {}
var readFile = options.readFile || ts.sys.readFile;
```

Then looked further into `ttypescript/lib/typescript.js` and found it was requiring `patch.js`. Then in `patch.js` I found the following:

```ts
// node_modules/ttypescript/lib/patch.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const path_1 = require("path");
const customRequire_1 = require("./customRequire");
console.log("1"); // happens
require('ts-node').register({ project: __dirname + '/../ts-node-config/tsconfig.json' });
console.log("2"); // doesn't happen
```

Finally, by commenting out the register line everything was working perfectly and my tests ran:

```ts
// node_modules/ttypescript/lib/patch.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const path_1 = require("path");
const customRequire_1 = require("./customRequire");
// require('ts-node').register({ project: __dirname + '/../ts-node-config/tsconfig.json' });
```

What does this register do? Is it necessary all the time? Is there a way we can change the code here to prevent the error in this circumstance? Thanks!

**Comment by cevek** [comment]
It happens because of circular dependency: ts-node try to require ttypescript, but ttypescript requires ts-node to compile transformers.

Fixed this problems in v1.0.9
Can you check it out in your environment?

**Comment by dsherret** [comment]
Hey @cevek, thanks for looking into this so fast! I found the fix works great on linux, but on windows since backslashes are annoyingly used for the path (ex. `\ts-node\`) it won't work.

**Comment by dsherret** [comment]
I can submit a fix right now actually.

**Comment by cevek** [comment]
PR welcome :)

**Comment by dsherret** [comment]
Closing because this is fixed in 1.0.10. Thanks again!
